packages/api-server/src/error.ts

```diff
@@ -0,0 +1,160 @@
+//  Error code from JSON-RPC 2.0 spec
+//  reference: http://www.jsonrpc.org/specification#error_object
+
+import { JSONRPCError } from "jayson";
+import { COMPATIBLE_DOCS_URL } from "./methods/constant";
+
+export interface ErrorKind {
+  code: number;
+  message: string;
+}
+
+export class AppError extends Error implements JSONRPCError {
+  code: number;
+  data?: object;
```

Comment:
```suggestion
  data?: CustomType;
```
can we restrict some type to replace `object`, the original JSONRPCError is not doing a very good job here.

Reply:
What do you think the `CustomType` looks like? Or what kinds of information do you think should be restricted in the "error.data"?

Reply:
simple thougt:

1. is there any typical pattern in using the `AppError` from our app? 
2. if not, we can at least use key-value type `[key: string]: any` to replace `object`.

Reply:
> 1. is there any typical pattern in using the AppError from our app?

```
AppError {
  code: <hardcoded error code>,
  message: <hardcoded sentence, demonstrates when happens or what happens>,
  data: <object type, provides enough context of the error>
}
```

> 2. if not, we can at least use key-value type [key: string]: any to replace object.

I choose `object` for 2 reason:
- `JSONRPCError` uses `data?: object`, so reusing `JSONRPCError` is nature for me
- `object` is simple then `Record` and `Map` for me

What's benefit for choosing `[key: string]: any`?

Reply:
https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/ban-types.md#default-options

Reply:
1. I mean, is there any typical pattern in using the `data` filed of AppError from our app?

Reply:
well, currently, I cannot describe the "typical pattern". I need some time to think.
what do you think about this question?

Reply:
ok, let's consider later